### ISSUE TITLE
[Frontend] Some events do not get rendered in front-end

### ISSUE THREAD

**Issue by titipata** [titipata/penn-events-calendar#147]
Some people noticed that the events are missing from the site. For example, we are missing an event from Roozbeh Kiani (here: https://go.activecalendar.com/upennmins). I checked the data and it is already scraped and stored properly (`event_index = 591`). Not sure why front-end does not render it.

**Comment by bluenex** [comment]
@titipata The parsed datetime seems to be wrong for some events. This makes it a backend problem as the frontend only reads the parsed date. See an image below:

![image](https://user-images.githubusercontent.com/11027706/69599174-b3df6400-103e-11ea-83ae-07fda98b97aa.png)

Look at `date` and `date_dt` field.

**Comment by bluenex** [comment]
I've investigated a bit more and found that no matter what format you specified (e.g. `dayfirst=False`) the `dateutil.parser` will try to achieve the parse by auto applying `dayfirst=True`. This makes our parsing unstable and some of the parsed date are wrong (due to the also inconsistent raw date format from scraped event data).

We can workaround this by apply some logic before returning the parsed date. For example the `dayfirst=False` and `dayfirst=True` should not return the same result, if it does something may be going wrong. 

```python 
In [28]: dateutil.parser.parse('20-11-2019', dayfirst=True).strftime('%d-%m-%Y')
Out[28]: '20-11-2019'

In [29]: dateutil.parser.parse('20-11-2019').strftime('%d-%m-%Y')
Out[29]: '20-11-2019'

In [30]: dateutil.parser.parse('12-04-2019', dayfirst=True).strftime('%d-%m-%Y')
Out[30]: '12-04-2019'

In [31]: dateutil.parser.parse('12-04-2019').strftime('%d-%m-%Y')
Out[31]: '04-12-2019'
```

**Comment by titipata** [comment]
This should be fixed in 1c9b6c8906f55480bc56c720c0d70d39f6f0a012
